### ISSUE TITLE
Formula parser does not allow `/` (forward slash char) after first member

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#6007]
Issue imported from https://tracker.freecad.org/view.php?id=4196

* **Reporter:** babaroga
* **Date submitted:** 11/18/2019
* **FreeCAD version:** 
* **Category:** Bug
* **Status:** confirmed
* **Tags:** expressions

# Original report text

If someone try to enter eg. `12/2` in any field for X, Y, Z, constrain in sketcher, parser will not allow to type `/` sign just after first member in expression. All other operations works as expected.
Everything works OK if entered as formula.

Issue is described in [Forum thread](https://forum.freecadweb.org/viewtopic.php?f=3&t=40995)

# Steps to reproduce

Try to change X, Y or Z position of eg. body with

`12/2` - does not work
`(12)/2` - does not work
`0+12/2` - works

# FreeCAD Info

```
OS: Manjaro Linux (KDE//usr/share/xsessions/plasma)
Word size of OS: 64-bit
Word size of FreeCAD: 64-bit
Version: 0.19.18753 (Git)
Build type: Release
Branch: master
Hash: c0a80e1a9a5683404fe506201d0ee74b28ea819d
Python version: 3.7.4
Qt version: 5.13.2
Coin version: 4.0.0a
OCC version: 7.4.0
Locale: English/United States (en_US)

```

# Other bug information

* **Priority:** normal
* **Severity:** minor
* **Category:** Bug
* **OS: Manjaro 18.1.3**
* **Platform:** Linux
* **Updated:** 2/6/2021

# Discussion from Mantis ticket

### Comment by ulrich1a 2019-12-31 16:07
The reported bug is not caused by FreeCAD-Expressions. Expressions are used in the dialog reachable via the blue `fx`-symbol:  

![image](https://user-images.githubusercontent.com/4140247/220650012-bf0bff10-6e76-4c04-8b40-db5c038d5dcf.png)

This dialog is not affected by the bug here and behaves different and uses different code. 
The bug is related to the input field of the Gui, which uses a different parser. The formula typed in the input field is gone after its calculation. 

Both usages uses "expressions" but are coded in different areas of FreeCAD.

Ulrich

---

### Comment by @0penBrain 2020-01-01 17:47
ulrich1a : would you be able to give a pointer to involved parser code in GitHub ?

**Comment by luzpaz** [comment]
Cleaned up this ticket. 

@0penBrain question, should i remove the Expressions tag since per Ulrich this isn't related to the expressions engine ?

**Comment by 0penBrain** [comment]
@luzpaz indeed this is the formula parser involved here, not the expression one.
Also in author description, "formula" is used instead of "expression".

**Comment by adrianinsaval** [comment]
can't reproduce, most likely fixed by #7124
